src/Raven.Server/Documents/Indexes/IndexStorage.cs

```diff
@@ -467,8 +467,19 @@ public IEnumerable<Slice> GetItemKeysFromCollectionThatReference(string collecti
                 using (DocumentIdWorker.GetLower(tx.InnerTransaction.Allocator, referenceKey, out var k))
                 using (var it = collectionTree.MultiRead(k))
                 {
-                    if (it.Seek(Slices.BeforeAllKeys) == false)
-                        yield break;
+                    if (lastItemId == null)
+                    {
+                        if (it.Seek(Slices.BeforeAllKeys) == false)
+                            yield break;
+                    }
+                    else
+                    {
+                        using (Slice.From(tx.InnerTransaction.Allocator, lastItemId, out var idSlice))
+                        {
+                            if (it.Seek(idSlice) == false)
+                                yield break;
+                        }
```

Comment:
Should the compare and move next go here?

Reply:
Since I changed the code, we now get the document id _before_ it was processed so there is no need to do that.